output.js

```diff
@@ -264,6 +272,47 @@ function link(config, text, target) {
     return kolorist.link(text, target);
 }
 
+/**
+ * Convert tabs indentation to two spaces.
+ * @param {string} str 
+ */
+function tabs2Spaces(str) {
+    return str.replace(/^\t+/, tabs => '  '.repeat(tabs.length));
+}
+
+/**
+ * Generate an excerpt of the location in the source around the
+ * specified position. 
+ * @param {*} config 
+ * @param {string} content Text content to generate the code frame of
+ * @param {number} lineNum zero-based line number
+ * @param {number} columnNum zero-based column number
+ * @param {number} before Number of lines to show before the marker
+ * @param {number} columnNum Number of lines to show after the marker
+ */
+function genCodeFrame(config, content, lineNum, columnNum, before, after) {
+    const lines = content.split('\n');
+    const startLine = Math.max(0, lineNum - before);
+    const endLine = Math.min(lines.length - 1, lineNum + after);
+    const maxDigits = String(endLine).length;
```

Comment:
A tiny bit mis-named: These are (in normal code) _characters_, not _digits_.

Reply:
Good catch! Renamed the variable :100: